### PR TITLE
add github actions builds

### PR DESCRIPTION
**Pull request by sebthom**

### CODE REVIEW

**Comment by mickaelistria** [comment]
At the moment, we're satisfied with Jenkins builds. What value would this extra build offer that makes it worth the extra maintenance?

**Comment by rubenporras** [comment]
for what is worth, I also think like Sebastian that it would be very nice to CI feedback without creating a PR to the source repo. Usually I create a PR just to get the CI feedback, even when changes are not ready.

File: org.eclipse.lsp4e.tests.mock/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: Mock Language Server to test LSP4E
 Bundle-SymbolicName: org.eclipse.lsp4e.tests.mock
-Bundle-Version: 0.14.4.qualifier
+Bundle-Version: 0.14.5.qualifier
```
Comment (mickaelistria) [inline org.eclipse.lsp4e.tests.mock/META-INF/MANIFEST.MF:5]: This shouldn't be necessary. Please revert and consider the version bump separately if really needed.

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,277 @@
+Eclipse Public License - v 2.0
```
Comment (mickaelistria) [inline LICENSE:1]: Would be nice to add it in a dedicated PR.

**Review by mickaelistria — CHANGES_REQUESTED** [review]

File: org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -2,7 +2,7 @@ Manifest-Version: 1.0
 Bundle-ManifestVersion: 2
 Bundle-Name: JDT Integration for LSP4E
 Bundle-SymbolicName: org.eclipse.lsp4e.jdt;singleton:=true
-Bundle-Version: 0.10.2.qualifier
+Bundle-Version: 0.10.3.qualifier
```
[the remark this answers is not in the document]
Comment (sebthom) [reply org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF:5]: That is the same issue we talked about in the other tm4e issue. it looks like the moment a file is modified in the repo root the build fails complaining about the versions not being bumped. I have no idea how to solve this.

File: LICENSE (hunk 2 above)
Comment (sebthom) [reply LICENSE:1]: I can make a separate PR, but thought a dedicated commit is sufficient because the change is clearly visible in the git history.

File: org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF (hunk 3 above)
[the remark this answers is not in the document]
Comment (mickaelistria) [reply org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF:5]: That's not how it's supposed to work. I'll investigate.

File: org.eclipse.lsp4e.tests.mock/META-INF/MANIFEST.MF (hunk 1 above)
Comment (mickaelistria) [reply org.eclipse.lsp4e.tests.mock/META-INF/MANIFEST.MF:5]: I cannot reproduce a failure with a local build. Please submit the PR without changing versions to see how it does behave on CI.

File: org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF (hunk 3 above)
[the remark this answers is not in the document]
Comment (sebthom) [reply org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF]: I reverted the version bumps, now the github action fails with
`Only qualifier changed for (org.eclipse.lsp4e.jdt/0.10.2.202207051450). Expected to have bigger x.y.z than what is available in baseline (0.10.2.202205031750)`  see https://github.com/sebthom/lsp4e/runs/7198694165

[the remark this answers is not in the document]
Comment (sebthom) [reply org.eclipse.lsp4e.jdt/META-INF/MANIFEST.MF]: @mickaelistria that's the info I needed. the gha build no longer creates a shallow clone and now succeeds without bumping the versions.

**Comment by sebthom** [comment]
@rubenporras I am getting test failures in github builds. Any ideas why? https://github.com/sebthom/lsp4e/runs/7600681404?check_suite_focus=true  I am using a similar github action config for building/testing TM4E where it works fine.

File: org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/LanguageServiceAccessorTest.java [hunk 4]
```diff
@@ -434,15 +434,15 @@ public void testSingletonLS() throws Exception {
 		var testFile1 = createFile(project, "shouldUseSingletonLS.lsp-singletonLS", "");
 		IDocument document1 = getDocument(testFile1);
 		assertNotNull(document1);
-		var languageServers = getLanguageServers(document1, MATCH_ALL);
+		var languageServers = getLanguageServers(document1, MATCH_ALL).get();
```
Comment (mickaelistria) [inline org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/LanguageServiceAccessorTest.java:437]: I don't think this should be necessary here. The choice of using a singleton LS doesn't require that the LanguageServer is ready; this shouldn't be something we need for the test to pass. If this is required for the test to pass, then it probably shows underlying code has some bug.

File: org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/TestUtils.java [hunk 5]
```diff
@@ -237,21 +237,28 @@ public static IEditorReference[] getEditors() {
 		return null;
 	}
 
+	public static boolean isGitHubActions() {
+		return "true".equals(System.getenv("GITHUB_ACTIONS"));
```
Comment (mickaelistria) [inline org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/TestUtils.java:241]:
```suggestion
		return Boolean.getBoolean("GITHUB_ACTIONS");
```

Comment (sebthom) [reply org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/TestUtils.java:241]: Will do. thanks!

File: org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/LanguageServiceAccessorTest.java (hunk 4 above)
Comment (sebthom) [reply org.eclipse.lsp4e.test/src/org/eclipse/lsp4e/test/LanguageServiceAccessorTest.java:437]: No, it is not necessary. I am currently just trying out things as I am getting a lot of very random test failures on GHA.

**Comment by akurtakov** [comment]
Is this smth someone still want to finish?

**Comment by sebthom** [comment]
Yes, the problem is there are still random build failures.

**Comment by sebthom** [comment]
I spent a lot of time trying to get the jobs running without test failures on Windows and Ubuntu but so far there are still random test failures. Only MacOS builds are stable so I commented out Windows/Ubuntu runners for now until we find a way to stabilize the tests. Even with only mac builds I think this is a valuable addition.

**Comment by mickaelistria** [comment]
The initial target branch `master` was deleted and replaced by `main`, so this PR got closed automatically. If this is still relevant, please re-create this PR targetting the `main` branch.
